src/model/client.ts

```diff
@@ -222,38 +241,56 @@ export class CommandoClient extends Client {
   /**
    * Handles when a user sends a message.
    *
-   * @param msg - Message object
+   * @param msg - CommandoMessage object
    */
-  private async onMessage(msg: Message): Promise<void> {
+  private async onMessage(msg: CommandoMessage): Promise<void> {
     if (msg.author.bot) {
       return
     }
 
-    if (!msg.content.startsWith(this.options.commandPrefix)) {
-      return
+    let prefix: string | undefined
+
+    if (typeof this.options.commandPrefix === 'string') {
+      if (!msg.content.startsWith(this.options.commandPrefix)) {
+        return
+      }
+
+      prefix = this.options.commandPrefix
+    } else {
+      for (const pref of this.options.commandPrefix) {
+        if (msg.content.startsWith(pref)) {
+          prefix = pref
+          break
+        }
+      }
+
+      if (prefix === undefined) {
+        return
+      }
     }
 
-    const withoutPrefix = msg.content.slice(this.options.commandPrefix.length)
+    const withoutPrefix = msg.content.slice(prefix.length)
     const split = withoutPrefix.split(' ')
     const commandName = split[0]
     const args = split.slice(1)
 
     const command = this.getCommandByNameOrAlias(commandName)
 
     if (!command) {
-      // TODO: Handle non commands, optionally
+      this.emit('invalidCommand', msg)
+
       return
     }
 
-    if (command.options.args) {
-      const commandoMessageWithArgs = new CommandoMessage(msg, command)
+    this.emit('commandMessage', msg, command, prefix)
```

Comment:
I think we could simplify this to just `this.emit('command', command, msg)`

The prefix is available via `msg.client.options.commandPrefix`

Reply:
> The prefix is available via `msg.client.options.commandPrefix`
Not if you are using multiple prefixes